xarray/core/common.py

```diff
@@ -576,7 +576,13 @@ def where(self, cond, other=None, drop=False):
             outobj = self
             outcond = cond
 
-        return outobj._where(outcond)
+        # preserve attributes
+        out = outobj._where(outcond)
+        out.attrs = self.attrs
+        if isinstance(out, Dataset):
```

Comment:
I just tried ``actual = ds.a.where(ds > 1)``and _where() returns a ``NotImplemented`` string